store/tikv/kv.go

```diff
@@ -109,6 +110,37 @@ type tikvStore struct {
 	etcdAddrs    []string
 	mock         bool
 	enableGC     bool
+
+	safePoint uint64
+	spTime    time.Time
+	spSession tidb.Session  // this is used to obtain safePoint from remote
+	spMutex   sync.RWMutex  // this is used to update safePoint and spTime
+	spMsg     chan struct{} // this is used to nofity when the store is closed
+}
+
+func (s *tikvStore) UpdateSPCache(cachedSP uint64, cachedTime time.Time) {
+	s.spMutex.Lock()
+	s.safePoint = cachedSP
+	s.spTime = cachedTime
+	s.spMutex.Unlock()
+}
+
+func (s *tikvStore) CheckVisibility(startTime uint64) error {
+	s.spMutex.RLock()
+	cachedSafePoint := s.safePoint
+	cachedTime := s.spTime
+	s.spMutex.RUnlock()
+	diff := time.Since(cachedTime)
```

Comment:
It's possible that `SafePointChecker` can't load data in 10s...
Then this check make read data from TiKV totally unavailable, what's worse, 
if CheckVisibility fail, we can't read and update safepoint, 
if we can't update safepoint, CheckVisibility fail again...

Reply:
Good Insight! I have resorted to ETCD for storing data.
Thanks!